tools/v8-prof/polyfill.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2012 the V8 project authors. All rights reserved.
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Google Inc. nor the names of its
+//       contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// Node polyfill
+var os = {
+  system: function(name, args) {
+    return require('child_process').execFileSync(
+        name, args, {encoding: 'utf8'}).toString();
+  }
+};
+var print = console.log;
+var read = function(fileName) {
+  return require('fs').readFileSync(fileName, 'utf8');
+};
+arguments = process.argv.slice(2);
+
+// Polyfill "readline()".
+var fs = require('fs');
+var fd = fs.openSync(arguments[0], 'rs');
+var buf = new Buffer(4096);
+var line = "";
+var readline = function() {
+  while (true) {
+    var line_break = line.indexOf('\n');
+    if (line_break !== -1) {
+      var res = line.slice(0, line_break);
+      line = line.slice(line_break + 1);
+      return res;
+    }
+    var bytes = fs.readSync(fd, buf, 0, buf.length);
+    line += buf.toString('utf8', 0, bytes);
+    if (line.length === 0) {
+      return false;
+    }
+  }
+}
```

Comment:
Just curious, is there a reason to use `var f = function() {` instead of `function f() {` in this file?  If there is, can you add a semicolon here? :-)